### PR TITLE
fix: preserve Healthcheck field in image config

### LINKED ISSUE #197: Healthcheck field silently dropped from image config
## Problem

When passing a `Healthcheck` field in the `config` attribute of `buildImage`, the field is silently dropped from the final OCI image. Other Docker-extension fields like `StopSignal` happen to work because they exist in the OCI `v1.ImageConfig` struct, but `Healthcheck` does not.

## Root Cause

In `cmd/image.go` (line 112), the config JSON is unmarshalled into `v1.ImageConfig` from the OCI image-spec:

```go
var imageConfig v1.ImageConfig
// ...
err = json.Unmarshal(imageConfigJson, &imageConfig)
```

The OCI `v1.ImageConfig` struct ([source](https://github.com/opencontainers/image-spec/blob/v1.1.1/specs-go/v1/config.go)) does **not** include a `Healthcheck` field — it's a Docker-specific extension. Go's `json.Unmarshal` silently ignores unknown fields, so the `Healthcheck` data is lost.

The same type is used in `types/types.go` line 19:

```go
type Image struct {
    // ...
    ImageConfig v1.ImageConfig `json:"image-config"`
    // ...
}
```

## Reproduction

```nix
nix2container.buildImage {
  name = "test";
  tag = "latest";
  config = {
    entrypoint = [ "${pkgs.coreutils}/bin/true" ];
    Healthcheck = {
      Test = [ "CMD" "curl" "-f" "http://localhost/health" ];
      Interval = 30000000000;
      Timeout = 5000000000;
      StartPeriod = 5000000000;
      Retries = 3;
    };
    StopSignal = "SIGQUIT";
  };
}
```

Inspecting the intermediate `config.json` (the Nix derivation input) confirms `Healthcheck` **is present**. But `skopeo inspect` on the built image shows it's missing. `StopSignal` is preserved because it happens to be an OCI-spec field.

## Suggested Fix

Extend `ImageConfig` in `types/types.go` to include the Docker `Healthcheck` field. For example:

```go
// ImageConfig extends the OCI ImageConfig with Docker-specific fields.
type ImageConfig struct {
    v1.ImageConfig

    // Healthcheck describes how to check the container is healthy (Docker extension).
    Healthcheck *HealthConfig `json:"Healthcheck,omitempty"`
}

// HealthConfig holds configuration for container health checks.
type HealthConfig struct {
    Test        []string `json:"Test,omitempty"`
    Interval    int64    `json:"Interval,omitempty"`
    Timeout     int64    `json:"Timeout,omitempty"`
    StartPeriod int64    `json:"StartPeriod,omitempty"`
    Retries     int      `json:"Retries,omitempty"`
}
```

Then update `cmd/image.go` to use `types.ImageConfig` instead of `v1.ImageConfig`.

This preserves backwards compatibility (all existing OCI fields still work via embedding) while adding support for the Docker `Healthcheck` extension that many container runtimes and tools (Docker, Podman, Trivy CIS compliance checks) rely on.

## Impact

- Any `Healthcheck` passed via `buildImage.config` is silently lost
- Container health checks configured in Nix never appear in the built image
- CIS Docker Benchmark check 4.6 ("Ensure HEALTHCHECK instructions have been added") always fails
- Tools like `docker inspect`, `podman inspect`, and Trivy compliance scans cannot see the healthcheck

### PR DESCRIPTION
**Pull request by Dauliac**
## Summary

- The `Healthcheck` field passed via `buildImage { config.Healthcheck = ...; }` was silently dropped from the final image
- `StopSignal`, `WorkingDir`, `Volumes` etc. worked because they exist in the OCI `v1.ImageConfig` struct, but `Healthcheck` is a Docker-specific extension not in the OCI spec
- Go's `json.Unmarshal` silently ignores unknown fields, so the data was lost

## Changes

- **`types/types.go`**: Add `HealthConfig` struct and `ImageConfig` type that embeds `v1.ImageConfig` + adds `Healthcheck`
- **`cmd/image.go`**: Unmarshal config.json into `types.ImageConfig` instead of `v1.ImageConfig`
- **`nix/image.go`**: `GetConfigBlob()` now serializes a `dockerImage` wrapper that preserves `Healthcheck` in the output config blob
- **`nix/image_test.go`**: Add tests verifying Healthcheck is present when set and absent when nil

## Backwards Compatibility

Fully backwards compatible:
- `types.ImageConfig` embeds `v1.ImageConfig`, so all existing OCI fields work identically
- When no `Healthcheck` is set (`nil`), the field is omitted from JSON output (`omitempty`)
- Existing Nix configs that don't use `Healthcheck` produce identical output

Fixes #197

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -11,15 +11,34 @@ import (
...
+	Test        []string `json:"Test,omitempty"`
+	Interval    int64    `json:"Interval,omitempty"`
+	Timeout     int64    `json:"Timeout,omitempty"`
+	StartPeriod int64    `json:"StartPeriod,omitempty"`
+	Retries     int      `json:"Retries,omitempty"`
+}
+
+// ImageConfig extends the OCI ImageConfig with Docker-specific fields
+// such as Healthcheck that are widely supported by container runtimes.
+type ImageConfig struct {
```
Comment (nlewo) [inline types/types.go:26]: Could you use https://github.com/moby/docker-image-spec/blob/main/specs-go/v1/image.go#L20 instead?

Comment (nlewo) [reply types/types.go:26]: And add a comment to this issue which talk about this issue: https://github.com/opencontainers/image-spec/issues/749

**Review by nlewo — CHANGES_REQUESTED** [review]
Pragmatically, i'm in favor of adding this.
